File: Gun2/2.React/2.eTicaret/eTicaretServer/Models/ShoppingCart.cs
namespace eTicaretServer.Models;

public sealed class ShoppingCart
{
    public ShoppingCart()
    {
        Id = Guid.NewGuid();
    }
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public Product? Product { get; set; }
    public int Quantity { get; set; }
}

File: Gun5/e-ticaret-next-app/models/shopping-cart.js
import mongoose from "mongoose";

const ShoppingCartSchema = new mongoose.Schema({
  productId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
  },
  quantity: {
    type: Number,
    required: true,
    min:1
  },
});

export default mongoose.models.ShoppingCart || mongoose.model("ShoppingCart", ShoppingCartSchema);